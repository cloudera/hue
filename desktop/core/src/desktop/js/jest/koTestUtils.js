// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as ko from 'knockout';

export const koSetup = () => {
  let originalLoadTemplate;
  let wrapper;

  beforeEach(() => {
    originalLoadTemplate = ko.components.defaultLoader.loadTemplate;
    ko.components.defaultLoader.loadTemplate = (name, templateConfig, callback) => {
      const div = window.document.createElement('div');
      div.innerHTML = templateConfig;
      callback(div.children);
    };

    wrapper = window.document.createElement('div');
    wrapper.classList.add('component-wrapper');
    window.document.querySelector('body').appendChild(wrapper);
  });

  afterEach(() => {
    ko.components.defaultLoader.loadTemplate = originalLoadTemplate;
    wrapper.parentNode.removeChild(wrapper);
    ko.cleanNode(wrapper);
    expect(window.document.querySelectorAll('[data-test]').length).toEqual(0);
  });

  return {
    renderComponent: async (name, data, instantTimeout) =>
      new Promise((resolve, reject) => {
        const element = window.document.createElement('div');
        element.setAttribute(
          'data-bind',
          `descendantsComplete: descendantsComplete, component: { name: "${name}", params: params }`
        );
        wrapper.appendChild(element);

        try {
          ko.components.get(name, comp => {
            if (comp.createViewModel) {
              const origSetTimeout = window.setTimeout;
              if (instantTimeout) {
                window.setTimeout = cb => cb();
                comp.createViewModel(data, { element: element });
                window.setTimeout = origSetTimeout;
              } else {
                comp.createViewModel(data, { element: element });
              }

              ko.applyBindings(
                {
                  descendantsComplete: () => resolve(wrapper),
                  params: data
                },
                wrapper
              );
            } else {
              reject('no createViewModel function found on component ' + name);
            }
          });
        } catch (e) {
          reject(e);
        }
      }),

    renderKo: async (html, viewModel) =>
      new Promise(resolve => {
        const setupContainer = window.document.createElement('div');
        setupContainer.setAttribute(
          'data-bind',
          '{ descendantsComplete: descendantsComplete, with: model }'
        );

        const container = window.document.createElement('div');
        container.innerHTML = html;
        setupContainer.appendChild(container);

        const data = {
          model: viewModel,
          descendantsComplete: () => resolve(container)
        };

        ko.applyBindings(data, setupContainer);
      }),

    waitForKoUpdate: async () =>
      new Promise(resolve => {
        window.setTimeout(resolve, 0);
      })
  };
};

export const waitForObservableChange = async observable =>
  new Promise(resolve => {
    const sub = observable.subscribe(() => {
      sub.dispose();
      resolve();
    });
  });

export const simulateInput = (element, text) => {
  element.value = text;
  element.dispatchEvent(new Event('change'));
};
