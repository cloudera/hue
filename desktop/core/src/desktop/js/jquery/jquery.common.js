// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import $ from 'jquery';
import 'jquery/jquery.migration';

import 'jquery-contextmenu'; // TODO: Remove, it's only used for "old" query builder
import 'jquery.cookie';
import 'jquery-form';
import 'jquery/plugins/jquery.selectize';
import 'selectize-plugin-clear';

import 'jquery-ui/ui/widgets/autocomplete';
import 'jquery-ui/ui/widgets/mouse';
import 'jquery-ui/ui/widgets/draggable';
import 'jquery-ui/ui/widgets/droppable';
import 'jquery-ui/ui/widgets/resizable';
import 'jquery-ui/ui/widgets/sortable';

import 'ext/jquery.chosen.1.0.0.min';
import 'ext/jquery.dataTables.1.8.2.min';
import 'ext/jquery.select2.3.5.1.min';

import 'jquery/plugins/jquery.datatables.sorting';
import 'jquery/plugins/jquery.delayedinput';
import 'jquery/plugins/jquery.filechooser';
import 'jquery/plugins/jquery.hdfs.autocomplete';
import 'jquery/plugins/jquery.hiveautocomplete';
import 'jquery/plugins/jquery.horizontalscrollbar';
import 'jquery/plugins/jquery.huedatatable';
import 'jquery/plugins/jquery.notify';
import 'jquery/plugins/jquery.rowselector';
import 'jquery/plugins/jquery.scrollleft';
import 'jquery/plugins/jquery.scrollup';
import 'jquery/plugins/jquery.selector';
import 'jquery/plugins/jquery.tableextender';
import 'jquery/plugins/jquery.tableextender2';
import 'jquery/plugins/jquery.tablescroller';
import 'jquery/plugins/jquery.titleupdater';

window.$ = $;
window.jQuery = $;

export default $;
