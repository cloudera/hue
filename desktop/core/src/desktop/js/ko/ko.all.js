// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as ko from 'knockout';
import 'ko/ko.init';
import komapping from 'knockout.mapping';
import 'knockout-switch-case';
import 'knockout-sortable';
import 'knockout.validation';

import 'ext/ko.editable.custom';
import 'ext/ko.selectize.custom';

import 'ko/bindings/charts/mapchart/ko.mapChart';
import 'ko/bindings/charts/ko.barChart';
import 'ko/bindings/charts/ko.leafletMapChart';
import 'ko/bindings/charts/ko.lineChart';
import 'ko/bindings/charts/ko.partitionChart';
import 'ko/bindings/charts/ko.pieChart';
import 'ko/bindings/charts/ko.scatterChart';
import 'ko/bindings/charts/ko.timelineChart';

import 'ko/bindings/ko.vue.js';

import 'ko/bindings/ace/ko.aceEditor';
import 'ko/bindings/ace/ko.aceResizer';
import 'ko/bindings/ko.appAwareTemplateContextMenu';
import 'ko/bindings/ko.assistFileDraggable';
import 'ko/bindings/ko.assistFileDroppable';
import 'ko/bindings/ko.assistVerticalResizer';
import 'ko/bindings/ko.attachViewModelToElementData';
import 'ko/bindings/ko.augmentHtml';
import 'ko/bindings/ko.autocomplete';
import 'ko/bindings/ko.autogrowInput';
import 'ko/bindings/ko.blurHide';
import 'ko/bindings/ko.bytesize';
import 'ko/bindings/ko.chosen';
import 'ko/bindings/ko.clearable';
import 'ko/bindings/ko.clickForAceFocus';
import 'ko/bindings/ko.clickToCopy';
import 'ko/bindings/ko.clipboard';
import 'ko/bindings/ko.codemirror';
import 'ko/bindings/ko.contextMenu';
import 'ko/bindings/ko.contextSubMenu';
import 'ko/bindings/ko.datalist';
import 'ko/bindings/ko.datepicker';
import 'ko/bindings/ko.dateRangePicker';
import 'ko/bindings/ko.dblClick';
import 'ko/bindings/ko.delayedOverflow';
import 'ko/bindings/ko.dockable';
import 'ko/bindings/ko.documentChooser';
import 'ko/bindings/ko.documentContextPopover';
import 'ko/bindings/ko.draggableText';
import 'ko/bindings/ko.dropdown';
import 'ko/bindings/ko.dropzone';
import 'ko/bindings/ko.duration';
import 'ko/bindings/ko.ellipsis';
import 'ko/bindings/ko.fadeVisible';
import 'ko/bindings/ko.fetchMore';
import 'ko/bindings/ko.fileChooser';
import 'ko/bindings/ko.foreachVisible';
import 'ko/bindings/ko.fresherEditor';
import 'ko/bindings/ko.hdfsAutocomplete';
import 'ko/bindings/ko.hdfsTree';
import 'ko/bindings/ko.highlight';
import 'ko/bindings/ko.hiveChooser';
import 'ko/bindings/ko.html';
import 'ko/bindings/ko.hueach';
import 'ko/bindings/ko.hueCheckAll';
import 'ko/bindings/ko.hueCheckbox';
import 'ko/bindings/ko.hueChecked';
import 'ko/bindings/ko.hueLink';
import 'ko/bindings/ko.hueSpinner';
import 'ko/bindings/ko.impalaDagre';
import 'ko/bindings/ko.jHueRowSelector';
import 'ko/bindings/ko.logResizer';
import 'ko/bindings/ko.logScroller';
import 'ko/bindings/ko.medium';
import 'ko/bindings/ko.moment';
import 'ko/bindings/ko.momentFromNow';
import 'ko/bindings/ko.multiCheck';
import 'ko/bindings/ko.multiCheckForeachVisible';
import 'ko/bindings/ko.multiClick';
import 'ko/bindings/ko.multiLineEllipsis';
import 'ko/bindings/ko.numberFormat';
import 'ko/bindings/ko.numericTextInput';
import 'ko/bindings/ko.onClickOutside';
import 'ko/bindings/ko.oneClickSelect';
import 'ko/bindings/ko.parseArguments';
import 'ko/bindings/ko.publish';
import 'ko/bindings/ko.reactWrapper';
import 'ko/bindings/ko.readOnlyAce';
import 'ko/bindings/ko.resizable';
import 'ko/bindings/ko.select2';
import 'ko/bindings/ko.simplesize';
import 'ko/bindings/ko.slider';
import 'ko/bindings/ko.slideVisible';
import 'ko/bindings/ko.solrChooser';
import 'ko/bindings/ko.spinEdit';
import 'ko/bindings/ko.splitDraggable';
import 'ko/bindings/ko.splitFlexDraggable';
import 'ko/bindings/ko.sqlContextPopover';
import 'ko/bindings/ko.stickVertical';
import 'ko/bindings/ko.storageContextPopover';
import 'ko/bindings/ko.stretchDown';
import 'ko/bindings/ko.tagEditor';
import 'ko/bindings/ko.tagsNotAllowed';
import 'ko/bindings/ko.templateContextMenu';
import 'ko/bindings/ko.templatePopover';
import 'ko/bindings/ko.textSqueezer';
import 'ko/bindings/ko.toggle';
import 'ko/bindings/ko.toggleOverflow';
import 'ko/bindings/ko.tooltip';
import 'ko/bindings/ko.truncatedText';
import 'ko/bindings/ko.typeahead';
import 'ko/bindings/ko.verticalSlide';
import 'ko/bindings/ko.visibleOnHover';

import 'ko/components/appSwitcher/ko.appSwitcher';
import 'ko/components/assist/ko.assistDashboardPanel';
import 'ko/components/assist/ko.assistDbPanel';
import 'ko/components/assist/ko.assistDocumentsPanel';
import 'ko/components/assist/ko.assistEditorContextPanel';
import 'ko/components/assist/ko.assistFunctionsPanel';
import 'ko/components/assist/ko.assistGitPanel';
import 'ko/components/assist/ko.assistHBasePanel';
import 'ko/components/assist/ko.assistLangRefPanel';
import 'ko/components/assist/ko.assistPanel';
import 'ko/components/assist/ko.assistSchedulePanel';
import 'ko/components/assist/ko.assistStoragePanel';
import 'ko/components/assist/ko.rightAssistPanel';
import 'ko/components/contextPopover/ko.contextPopover';
import 'ko/components/contextPopover/ko.quickQueryContext';
import 'ko/components/contextPopover/ko.popoverOnHover';
import 'ko/components/doc/ko.shareDocModal';
import 'ko/components/simpleAceEditor/ko.simpleAceEditor';

import 'ko/components/ko.catalogEntriesList';
import 'ko/components/ko.contextSelector';
import 'ko/components/ko.createDirectoryModal';
import 'ko/components/ko.deleteDocModal';
import 'ko/components/ko.dropDown';
import 'ko/components/ko.editorDroppableMenu';
import 'ko/components/ko.executionAnalysis';
import 'ko/components/ko.favoriteApp';
import 'ko/components/ko.fieldSamples';
import 'ko/components/ko.globalSearch';
import 'ko/components/ko.historyPanel';
import 'ko/components/ko.importDocumentsModal';
import 'ko/components/ko.inlineAutocomplete';
import 'ko/components/ko.jobBrowserLinks';
import 'ko/components/ko.navProperties';
import 'ko/components/ko.navTags';
import 'ko/components/ko.pollingCatalogEntriesList';
import 'ko/components/ko.quickQueryAction';
import 'ko/components/ko.sentryPrivileges';
import 'ko/components/ko.sessionAuthModal';
import 'ko/components/ko.sessionPanel';
import 'ko/components/ko.shareGistModal';
import 'ko/components/ko.sqlColumnsTable';

// TODO: Move to about app when it has it's own webpack entry
import 'apps/about/components/ko.hueConfigTree';
import 'apps/about/components/ko.connectorsConfig';

import 'ko/extenders/ko.maxLength';
import 'ko/extenders/ko.numeric';
import 'ko/extenders/ko.toJson';

import 'ko/observables/ko.observableArrayDefault';
import 'ko/observables/ko.observableDefault';

window.ko = ko;
window.ko.mapping = komapping;

export default ko;
