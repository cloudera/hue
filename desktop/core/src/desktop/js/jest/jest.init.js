// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Ensure singletons
import 'core-js/stable';
import 'regenerator-runtime/runtime';

import 'apps/editor/execution/sessionManager';
import './jquery.setup';
import './sqlTestUtils';

import 'ext/bootstrap.2.3.2.min';
import axios from 'axios';
import $ from 'jquery';
import * as ko from 'knockout';
import komapping from 'knockout.mapping';

ko.mapping = komapping;

class Tooltip {}

class AceRange {}

class Autocomplete {}

const globalVars = {
  ko: ko,
  AUTOCOMPLETE_TIMEOUT: 1,
  CACHEABLE_TTL: 1,
  HAS_LINK_SHARING: true,
  HAS_SQL_ANALYZER: false,
  HUE_I18n: {},
  HUE_BASE_URL: '',
  HUE_CHARTS: {
    TYPES: 'bar'
  },
  LOGGED_USERNAME: 'foo',
  LOGGED_USER_ID: 'bar',
  PREVENT_AUTOFILL_INPUT_ATTRS:
    'autocorrect="off" autocomplete="do-not-autocomplete" autocapitalize="off" spellcheck="false"',
  STATIC_URLS: {
    'impala/art/icon_impala_48.png': 'impala/art/icon_impala_48.png',
    'beeswax/art/icon_beeswax_48.png': 'beeswax/art/icon_beeswax_48.png'
  },
  SQL_COLUMNS_KNOWN_FACET_VALUES: {
    type: { array: -1, boolean: -1 }
  },
  ace: {
    edit: () => ({
      setOptions: () => {},
      getSession: () => ({
        setMode: () => {},
        doc: {
          createAnchor: () => ({
            on: () => {},
            getPosition: () => ({
              row: 0
            }),
            setPosition: () => {},
            detach: () => {}
          })
        },
        getTextRange: () => {},
        addGutterDecoration: () => {},
        removeGutterDecoration: () => {}
      }),
      setTheme: () => {},
      getValue: () => '',
      getSelectionRange: () => ({ start: { row: 0, column: 0 }, end: { row: 0, column: 0 } }),
      on: () => {},
      off: () => {},
      commands: {
        on: () => {},
        off: () => {}
      },
      container: {
        addEventListener: () => {},
        removeEventListener: () => {}
      }
    }),
    require: () => ({
      Tooltip: Tooltip,
      Range: AceRange,
      Autocomplete: Autocomplete
    })
  }
};

Object.keys(globalVars).forEach(key => {
  global[key] = globalVars[key];
  global.window[key] = globalVars[key];
});

$.ajaxSetup({
  beforeSend: function () {
    console.warn('actual jQuery ajax called');
    console.trace();
  }
});

axios.interceptors.request.use(config => {
  console.warn('Actual axios ajax request made to url: ' + config.url);
  console.trace();
  return config;
});

process.on('unhandledRejection', err => {
  fail(err);
});

jest.mock('../utils/i18nReact');
jest.mock('../utils/hueAnalytics');
