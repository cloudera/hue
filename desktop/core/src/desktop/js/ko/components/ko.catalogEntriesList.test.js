// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import $ from 'jquery';
import { koSetup } from 'jest/koTestUtils';
import { NAME } from './ko.catalogEntriesList';

import 'ko/components/ko.inlineAutocomplete';
import 'ko/components/ko.fieldSamples';
import 'ko/components/ko.dropDown';

describe('ko.catalogEntriesList.js', () => {
  const setup = koSetup();

  it('should render component', async () => {
    const element = await setup.renderComponent(NAME, {
      catalogEntry: {
        isField: () => true,
        isComplex: () => false,
        isDatabase: () => false,
        getChildren: () => $.Deferred().resolve([]),
        getSample: () => $.Deferred().reject(),
        getDialect: () => 'impala',
        loadNavigatorMetaForChildren: () => $.Deferred().reject(),
        loadOptimizerPopularityForChildren: () => $.Deferred().reject(),
        isTableOrView: () => false,
        isSource: () => false
      }
    });

    expect(element.innerHTML).toMatchSnapshot();
  });
});
