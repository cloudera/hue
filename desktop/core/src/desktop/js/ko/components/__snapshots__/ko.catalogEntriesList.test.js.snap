// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ko.catalogEntriesList.js should render component 1`] = `
"<div data-bind=\\"descendantsComplete: descendantsComplete, component: { name: &quot;catalog-entries-list&quot;, params: params }\\"><script type=\\"text/html\\" id=\\"entries-table-td-description\\">
    <td data-bind=\\"attr: { 'title': comment }\\">
    <!-- ko if: $parent.editableDescriptions && !window.HAS_READ_ONLY_CATALOG -->
      <div class=\\"hue-catalog-entries-table-desc\\" data-bind=\\"visibleOnHover: { selector: '.editable-inline-action' }\\">
        <div data-bind=\\"editable: comment, editableOptions: {
          mode: 'inline',
          enabled: true,
          type: 'textarea',
          showbuttons: 'bottom',
          inputclass: 'hue-table-browser-desc-input',
          toggle: 'manual',
          toggleElement: '.toggle-editable',
          placeholder: 'Add a description...',
          emptytext: 'Add a description...',
          inputclass: 'hue-catalog-entries-table-desc-input',
          rows: 6,
          save: saveComment,
          inlineEditAction: { editClass: 'toggle-editable editable-inline-action' },
          multiLineEllipsis: { overflowHeight: '40px', expandable: true, expandActionClass: 'editable-inline-action' }
        }\\">Add a description...</div>
      </div>
    <!-- /ko -->
    <!-- ko if: !$parent.editableDescriptions || window.HAS_READ_ONLY_CATALOG -->
      <div class=\\"entries-table-description\\" data-bind=\\"text: comment, multiLineEllipsis\\"></div>
    <!-- /ko -->
    </td>
  </script><script type=\\"text/html\\" id=\\"entries-table-tbody-no-entries\\">
    <tbody>
      <tr>
        <td style=\\"font-style: italic;\\" data-bind=\\"attr: { 'colspan': colCount + (typeof $component.selectedEntries !== 'undefined' ? 1 : 0) + ($component.contextPopoverEnabled ? 1 : 0) }\\">
          <!-- ko ifnot: hasErrors -->
          No entries found
          <!-- /ko -->
          <!-- ko if: hasErrors -->
          Error loading entries
          <!-- /ko -->
        </td>
      </tr>
    </tbody>
  </script><script type=\\"text/html\\" id=\\"entries-table-shared-headers\\">
    <!-- ko if: typeof selectedEntries !== 'undefined' -->
    <th width=\\"1%\\" class=\\"select-column\\"><div class=\\"hue-checkbox fa\\" data-bind=\\"hueCheckAll: { allValues: filteredEntries, selectedValues: selectedEntries }\\"></div></th>
    <!-- /ko -->
    <!-- ko if: contextPopoverEnabled -->
    <th width=\\"1%\\">&nbsp;</th>
    <!-- /ko -->
  </script><script type=\\"text/html\\" id=\\"entries-table-shared-columns\\">
    <!-- ko if: typeof $parent.selectedEntries !== 'undefined' -->
    <td width=\\"1%\\" class=\\"select-column\\"><div class=\\"hue-checkbox fa\\" data-bind=\\"multiCheck: '#entryTable', value: $data, hueChecked: $parent.selectedEntries\\"></div></td>
    <!-- /ko -->
    <!-- ko if: $parent.contextPopoverEnabled -->
    <td width=\\"1%\\"><a href=\\"javascript: void(0);\\" data-bind=\\"click: showContextPopover\\"><i class=\\"fa fa-info\\"></i></a></td>
    <!-- /ko -->
  </script><div class=\\"context-popover-inline-autocomplete\\">
    <div class=\\"context-popover-sample-filter\\">
      <!-- ko component: {
        name: 'inline-autocomplete',
        params: {
          querySpec: querySpec,
          facets: facets,
          knownFacetValues: knownFacetValues,
          autocompleteFromEntries: autocompleteFromEntries
        }
      } --><div class=\\"inline-autocomp-container\\">
    <div>
      <!-- ko if: showMagnify --><!-- /ko-->
      <form autocomplete=\\"off\\">
        <input class=\\"inline-autocomp-input\\" autocorrect=\\"off\\" autocomplete=\\"do-not-autocomplete\\" autocapitalize=\\"off\\" spellcheck=\\"false\\" type=\\"text\\" data-bind=\\"
          attr: { 'placeHolder' : hasFocus() ? '' : placeHolder },
          textInput: searchInput,
          hasFocus: hasFocus,
          clearable: { value: searchInput, onClear: onClear },
          css: { 'inline-autocomp-magnify-input': showMagnify }\\" placeholder=\\"Filter...\\">
        <input class=\\"inline-autocomp-autocomp\\" disabled=\\"\\" type=\\"text\\" autocomplete=\\"do-not-autocomplete\\" data-bind=\\"
          value: inlineAutocomplete,
          css: { 'inline-autocomp-magnify-input': showMagnify }\\">
      </form>
    </div>
  </div><div class=\\"hue-drop-down-container hue-drop-down-fixed\\" data-bind=\\"event: { 'mousedown': facetDropDownMouseDown }, css: { 'open' : facetDropDownVisible() }, dropDownKeyUp: { onEsc: facetDropDownOnEsc, onEnter: facetDropDownOnEnter, onSelected: facetDropDownOnSelected, dropDownVisible: facetDropDownVisible }\\">
    <div class=\\"dropdown-menu\\" style=\\"overflow-y: auto; display: none;\\" data-bind=\\"visible: facetSuggestions().length > 0\\">
      <ul class=\\"hue-inner-drop-down\\" data-bind=\\"foreach: facetSuggestions\\"></ul>
    </div>
  </div><!-- /ko -->
    </div>
  </div><div class=\\"catalog-entries-list-container\\">
    <!-- ko hueSpinner: { spin: loading, center: true, size: 'xlarge' } --><!-- /ko -->
    <!-- ko if: !loading() && catalogEntry().isSource() --><!-- /ko -->

    <!-- ko if: !loading() && catalogEntry().isDatabase() --><!-- /ko -->

    <!-- ko if: !loading() && (catalogEntry().isTableOrView() || catalogEntry().isComplex()) --><!-- /ko -->

    <!-- ko if: !loading() && catalogEntry().isField() && !catalogEntry().isComplex() -->
    <!-- ko component: { name: 'field-samples', params: {
        catalogEntry: catalogEntry,
        onSampleClick: onSampleClick,
        refreshSampleInterval: refreshSampleInterval
      }} --><div class=\\"context-popover-inline-autocomplete\\" style=\\"display: flex\\">
    <div class=\\"context-popover-sample-filter\\">
      <!-- ko component: {
        name: 'inline-autocomplete',
        params: {
          querySpec: querySpec,
          autocompleteFromEntries: autocompleteFromEntries
        }
      } --><div class=\\"inline-autocomp-container\\">
    <div>
      <!-- ko if: showMagnify --><!-- /ko-->
      <form autocomplete=\\"off\\">
        <input class=\\"inline-autocomp-input\\" autocorrect=\\"off\\" autocomplete=\\"do-not-autocomplete\\" autocapitalize=\\"off\\" spellcheck=\\"false\\" type=\\"text\\" data-bind=\\"
          attr: { 'placeHolder' : hasFocus() ? '' : placeHolder },
          textInput: searchInput,
          hasFocus: hasFocus,
          clearable: { value: searchInput, onClear: onClear },
          css: { 'inline-autocomp-magnify-input': showMagnify }\\" placeholder=\\"Filter...\\">
        <input class=\\"inline-autocomp-autocomp\\" disabled=\\"\\" type=\\"text\\" autocomplete=\\"do-not-autocomplete\\" data-bind=\\"
          value: inlineAutocomplete,
          css: { 'inline-autocomp-magnify-input': showMagnify }\\">
      </form>
    </div>
  </div><div class=\\"hue-drop-down-container hue-drop-down-fixed\\" data-bind=\\"event: { 'mousedown': facetDropDownMouseDown }, css: { 'open' : facetDropDownVisible() }, dropDownKeyUp: { onEsc: facetDropDownOnEsc, onEnter: facetDropDownOnEnter, onSelected: facetDropDownOnSelected, dropDownVisible: facetDropDownVisible }\\">
    <div class=\\"dropdown-menu\\" style=\\"overflow-y: auto; display: none;\\" data-bind=\\"visible: facetSuggestions().length > 0\\">
      <ul class=\\"hue-inner-drop-down\\" data-bind=\\"foreach: facetSuggestions\\"></ul>
    </div>
  </div><!-- /ko -->
    </div>
    <!-- ko if: showOperations -->
    <div class=\\"context-popover-sample-controls\\">
      <div class=\\"margin-left-10 inline-block\\" data-bind=\\"component: { name: 'hue-drop-down', params: { value: operation, entries: operations } }\\"><a class=\\"inactive-action hue-drop-down-active\\" href=\\"javascript:void(0)\\" data-bind=\\"toggle: dropDownVisible, css: { 'blue': dropDownVisible }\\">
    <!-- ko if: icon --><!-- /ko -->
    <!-- ko if: !noLabel && value -->
    <span class=\\"hue-drop-down-selected\\" data-bind=\\"text: value() &amp;&amp; typeof value()[labelAttribute] !== 'undefined' ? value()[labelAttribute] : value(), visible: ! dropDownVisible() || !searchable, attr: { 'title': value() &amp;&amp; typeof value()[labelAttribute] !== 'undefined' ? value()[labelAttribute] : value() }\\" title=\\"default\\">default</span>
    <!-- /ko -->
    <i class=\\"fa fa-caret-down\\"></i>
  </a><input class=\\"hue-drop-down-input\\" type=\\"text\\" data-bind=\\"textInput: filter, attr: { 'placeHolder': inputPlaceHolder }, visible: dropDownVisible, style: { color: filterEdited() ? '#000' : '#AAA', 'min-height': '22px', 'margin-left': '10px' }\\" placeholder=\\"default\\" style=\\"display: none; color: rgb(170, 170, 170); min-height: 22px; margin-left: 10px;\\"><i class=\\"fa fa-caret-down\\"></i><div class=\\"hue-drop-down-container\\" data-bind=\\"css: { 'open' : dropDownVisible, 'hue-drop-down-fixed': fixedPosition, 'hue-drop-down-container-searchable': searchable }, dropDownKeyUp: { onEsc: onEsc, onEnter: onEnter, dropDownVisible: dropDownVisible }\\">
    <div style=\\"overflow-y: auto;\\" class=\\"dropdown-menu\\" data-bind=\\"visible: filteredEntries().length > 0\\">
      <!-- ko if: foreachVisible --><!-- /ko -->
      <!-- ko ifnot: foreachVisible -->
      <ul class=\\"hue-inner-drop-down\\" data-bind=\\"foreach: filteredEntries\\">
        <!-- ko if: typeof $data.divider !== 'undefined' && $data.divider --><!-- /ko -->
        <!-- ko if: typeof $data.divider === 'undefined' || !$data.divider -->
        <li><a href=\\"javascript:void(0)\\" data-bind=\\"text: $data &amp;&amp; typeof $data[$parent.labelAttribute] !== 'undefined' ? $data[$parent.labelAttribute] : $data, click: function () { let previous = $parent.value(); $parent.value($data); $parent.onSelect($data, previous); }\\">default</a></li>
        <!-- /ko -->
      
        <!-- ko if: typeof $data.divider !== 'undefined' && $data.divider --><!-- /ko -->
        <!-- ko if: typeof $data.divider === 'undefined' || !$data.divider -->
        <li><a href=\\"javascript:void(0)\\" data-bind=\\"text: $data &amp;&amp; typeof $data[$parent.labelAttribute] !== 'undefined' ? $data[$parent.labelAttribute] : $data, click: function () { let previous = $parent.value(); $parent.value($data); $parent.onSelect($data, previous); }\\">distinct</a></li>
        <!-- /ko -->
      
        <!-- ko if: typeof $data.divider !== 'undefined' && $data.divider --><!-- /ko -->
        <!-- ko if: typeof $data.divider === 'undefined' || !$data.divider -->
        <li><a href=\\"javascript:void(0)\\" data-bind=\\"text: $data &amp;&amp; typeof $data[$parent.labelAttribute] !== 'undefined' ? $data[$parent.labelAttribute] : $data, click: function () { let previous = $parent.value(); $parent.value($data); $parent.onSelect($data, previous); }\\">max</a></li>
        <!-- /ko -->
      
        <!-- ko if: typeof $data.divider !== 'undefined' && $data.divider --><!-- /ko -->
        <!-- ko if: typeof $data.divider === 'undefined' || !$data.divider -->
        <li><a href=\\"javascript:void(0)\\" data-bind=\\"text: $data &amp;&amp; typeof $data[$parent.labelAttribute] !== 'undefined' ? $data[$parent.labelAttribute] : $data, click: function () { let previous = $parent.value(); $parent.value($data); $parent.onSelect($data, previous); }\\">min</a></li>
        <!-- /ko -->
      </ul>
      <!-- /ko -->
    </div>
  </div></div>
      <div class=\\"margin-left-10 inactive-action inline-block\\">
        <!-- ko if: loadingSamples() || refreshSampleEnabled() --><!-- /ko -->
        <!-- ko if: !loadingSamples() && !refreshSampleEnabled() -->
        <a href=\\"javascript:void(0);\\" data-bind=\\"click: function () { loadSamples(true) }\\"><i class=\\"fa fa-play\\"></i></a>
        <!-- /ko -->
      </div>
    </div>
    <!-- /ko -->
  </div><table class=\\"table table-condensed table-nowrap\\">
    <thead>
    <tr>
      <th>Sample</th>
    </tr>
    </thead>
    <!-- ko if: loadingSamples() && (!refreshSampleEnabled() || (refreshSampleEnabled() && columnSamples().length === 0))  --><!-- /ko -->

    <!-- ko if: !loadingSamples() || (refreshSampleEnabled() && columnSamples().length > 0)  -->
    <tbody data-bind=\\"foreach: filteredColumnSamples\\"></tbody>
    <!-- ko if: filteredColumnSamples().length === 0 -->
    <tbody>
    <tr>
      <!-- ko ifnot: hasErrors --><!-- /ko -->
      <!-- ko if: hasErrors -->
      <td style=\\"font-style: italic;\\">Error loading samples</td>
      <!-- /ko -->
    </tr>
    </tbody>
    <!-- /ko -->
    <!-- /ko -->
  </table><!-- /ko -->
    <!-- /ko -->
  </div></div>"
`;
